src/Stratis.Bitcoin/P2P/Protocol/Payloads/ProvenHeadersPayload.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.P2P.Protocol.Payloads
+{
+    /// <summary>
+    /// Proven block headers received after a getheaders message.
+    /// </summary>
+    [Payload("provhdr")]
+    public class ProvenHeadersPayload : Payload
+    {
+        private List<BlockHeader> headers = new List<BlockHeader>();
+
+        public List<BlockHeader> Headers => this.headers;
+
+        public ProvenHeadersPayload()
+        {
+        }
+
+        public ProvenHeadersPayload(params ProvenBlockHeader[] headers)
```

Comment:
why params?

Reply:
That's how Headers payload doing in. I am following the same convention to make it consistent. Might be handy so you can either pass one header or multiple.